create_key/create_key.go

```diff
@@ -0,0 +1,147 @@
+	// we check for them.
+	err := Args.Parse(args[1:])
+	if err != nil {
+		return fmt.Errorf("could not parse arguments: %s", err)
+	}
+
+	// Args() returns the non-flag arguments, which we assume is the key
+	// filename. If more than one name is given, an error is returned.
+	if len(Args.Args()) > 1 {
+		return fmt.Errorf("unknown arguments: %v, expected a single filename", strings.Join(Args.Args(), ", "))
+	}
+	if len(Args.Args()) < 1 {
+		return errors.New("no filename given")
+	}
+	basename := Args.Args()[0]
+
+	// If the `outdir` flag is set, change to the output directory before
+	// writing the key files.
+	startDir := ""
+	if *outDir != "" {
+		startDir, err = os.Getwd()
+		if err != nil {
+			return fmt.Errorf("could not get current directory: %v", err)
+		}
+		err = os.Chdir(*outDir)
+		if err != nil {
+			return fmt.Errorf("could not enter output directory: %v", err)
+		}
+	}
+
+	// Write the key files
+	err = generateKeyPair(basename, "")
+	if err != nil {
+		return err
+	}
+
+	// If we changed to a different directory for writing files, we change back
+	// in case some other function is called later which assumes the working
+	// directory hasn't changed.
+	if *outDir != "" {
+		err = os.Chdir(startDir)
+		if err != nil {
+			return fmt.Errorf("could not return to start directory: %v", err)
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+// Generates a crypt4gh key pair, and saves it to the `<basename>.pub.pem` and
+// `<basename>.sec.pem` files. If any of the files already exists, the function
+// will instead return an error.
+func generateKeyPair(basename, password string) error {
+	privateKeyName := fmt.Sprintf("%s.sec.pem", basename)
+	publicKeyName := fmt.Sprintf("%s.pub.pem", basename)
+
+	// check if any of the files exist
+	if helpers.FileExists(publicKeyName) {
+		return fmt.Errorf("public key file %s already exists", publicKeyName)
+	}
+	if helpers.FileExists(privateKeyName) {
+		return fmt.Errorf("private key file %s already exists", privateKeyName)
+	}
+
+	// Generate key pair
+	log.Infof("Generating key pair: %s, %s", privateKeyName, publicKeyName)
+
+	publicKeyData, privateKeyData, err := keys.GenerateKeyPair()
+	if err != nil {
+		return err
+	}
+
+	// Save keys to file
+	pubFile, err := os.OpenFile(filepath.Clean(publicKeyName), os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := pubFile.Close(); err != nil {
+			log.Errorf("Error closing file: %s\n", err)
+		}
+	}()
+	err = keys.WriteCrypt4GHX25519PublicKey(pubFile, publicKeyData)
+	if err != nil {
+		return err
+	}
+
+	secFile, err := os.OpenFile(filepath.Clean(privateKeyName), os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := secFile.Close(); err != nil {
+			log.Errorf("Error closing file: %s\n", err)
+		}
+	}()
+	pass := []byte(password)
+	err = keys.WriteCrypt4GHX25519PrivateKey(secFile, privateKeyData, pass)
+
```

Comment:
Why don't you use the key generation function in the C4GH library?
https://github.com/elixir-oslo/crypt4gh/blob/b28572b08f0a3dd0a9cebb905ef2a563f8c3a0fa/main.go#L268

Reply:
they're not exported :/

Reply:
It is if you target the proper submodule
https://github.com/elixir-oslo/crypt4gh/blob/b28572b08f0a3dd0a9cebb905ef2a563f8c3a0fa/keys/keys.go#L44

Reply:
Or you can just duplicate the code in the main.go from the library.

Reply:
I could - but I honestly don't want to. :) I don't want this script to be interactive as it's likely to be used by scripts, and getting the function options from flag is just hard to read imho.

Reply:
Maybe not the code it self but the concept, currently you have duplicated every action.